qualtran/_infra/data_types.py

```diff
@@ -521,46 +521,120 @@ class QFxp(QDType):
 
     A real number can be approximately represented in fixed point using `num_int`
     bits for the integer part and `num_frac` bits for the fractional part. If the
-    real number is signed we require an additional bit to store the sign (0 for
-    +, 1 for -). In total there are `bitsize = (n_sign + num_int + num_frac)` bits used
-    to represent the number. E.g. Using `(bitsize = 8, num_frac = 6, signed = False)`
-    then $\pi$ \approx 3.140625 = 11.001001, where the . represents the decimal place.
+    real number is signed we store negative values in two's complement form. The first
+    bit can therefore be treated as the sign bit in such cases (0 for +, 1 for -).
+    In total there are `bitsize = (num_int + num_frac)` bits used to represent the number.
+    E.g. Using `(bitsize = 8, num_frac = 6, signed = False)` then
+    $\pi$ \approx 3.140625 = 11.001001, where the . represents the decimal place.
 
     We can specify a fixed point real number by the tuple bitsize, num_frac and
-    signed, with num_int determined as `(bitsize - num_frac - n_sign)`.
+    signed, with num_int determined as `(bitsize - num_frac)`.
```

Comment:
There's some docstrings and comments hiding around the code. Can you add one or two short paragraphs to the class docstring for QFxp to describe the relationship to the classical simulation protocol and how we'll use these "raw ints" in the classical simulation protocol.

Comments don't show up in the docs and may be missed, and docstrings on private methods may be missed. You can also add a line to the docstrings of the public methods that says something like 'see the class docstring for details on the classical simulation format'

Reply:
Updated the class docstring